pages/app/presenters/refinery/pages/content_page_presenter.rb

```diff
@@ -5,17 +5,22 @@ module Pages
     # and it will build sections from the page's parts. The page is not retained
     # internally, so if the page changes, you need to rebuild this ContentPagePresenter.
     class ContentPagePresenter < ContentPresenter
-      def initialize(page, page_title)
+      def initialize(page, page_title, current_refinery_user)
         super()
+        @current_refinery_user = current_refinery_user
+        @page = page
+
         add_default_title_section(page_title) if page_title.present? && Pages.show_title_in_body
         add_page_parts(page.parts) if page
         add_default_post_page_sections
       end
 
-    private
+      private
 
       def add_default_title_section(title)
-        add_section TitleSectionPresenter.new(:id => :body_content_title, :fallback_html => title)
+        Rails.cache.fetch([::I18n.locale, @current_refinery_user, @page, title]) do
```

Comment:
previously these presenters had no association with `Rails`, nor to users or anything like that. I'm uncomfortable adding those when caching can happen in the layer that has these.

Reply:
I understand, how should we handle this?

Reply:
Is it something we need to worry about / maintain?

Reply:
I think you're right, we should not worry about it right now:
* cache_pages_full_caching exists
* when a template is more complex, we don't fully use the `ContentPagePresenter`, so we can easily add Rais fragment cache per project.

After digging into this behavior, i think it's probably the admin pages#index who need a better cache: https://github.com/refinery/refinerycms/blob/master/pages/app/views/refinery/admin/pages/_page.html.erb